js/src/admin/components/ExtensionSettingsPage.tsx

```diff
@@ -108,7 +139,109 @@ export default class TrelloSettingsPage extends ExtensionPage {
     m.redraw();
   }
 
+  getTag(id: string): Tag | undefined {
+    return this.states.tags?.find((tag) => tag.id() === String(id));
+  }
+
+  addNewTagMappingButton(): JSX.Element {
+    return (
+      <Button
+        disabled={this.loading}
+        class="Button"
+        onclick={() =>
+          app.modal.show(NewTagMappingModal, {
+            states: this.states,
+            onClose: (boardShortLink: string, label: {}, tagId: string) => {
+              if (this.states.mappings[boardShortLink] === undefined) {
+                this.states.mappings[boardShortLink] = [];
+              }
+
+              const exists = this.states.mappings[boardShortLink].some((value) => value.tagId == tagId || value.label.id == label.id);
+
+              if (!exists) {
+                this.states.mappings[boardShortLink].push({ label, tagId });
+
+                this.states.dirty = true;
+                m.redraw();
+              }
+            },
+          })
+        }
+      >
+        {app.translator.trans('blomstra-trello.admin.settings.mapping.add_new')}
+      </Button>
+    );
+  }
+
+  saveButton(): Mithril.Children {
+    return (
+      <Button
+        disabled={this.states.loading || (!this.states.dirty && !this.dirty())}
+        class="Button Button--primary SaveButton"
+        onclick={async () => {
+          this.states.dirty = false;
+
+          await saveSettings({
+            'blomstra-trello.label-tag-mappings': JSON.stringify(this.states.mappings),
+            ...(this.dirty() ? this.dirty() : {}),
+          });
+        }}
+      >
+        {app.translator.trans('blomstra-trello.admin.settings.save')}
+      </Button>
+    );
+  }
+
+  private async loadFlarumTags() {
+    await app.store.find('tags', { include: 'parent' });
+
+    this.states.tags = sortTags(app.store.all('tags'));
+  }
+
+  private loadMappings() {
+    try {
+      this.states.mappings = JSON.parse(app.data.settings['blomstra-trello.label-tag-mappings']);
+
+      if (!this.states.mappings) {
+        this.states.mappings = {};
+      }
+    } catch {
+      this.states.mappings = {};
+    }
+  }
+
   content() {
+    function displayLabel(label: {}) {
+      let attrs = {};
+
+      // Trello color choices
+      const colors = {
+        green: '#61bd4f',
+        yellow: '#f2d600',
+        orange: '#ff9f1a',
+        red: '#eb5a46',
+        purple: '#c377e0',
+        blue: '#0079bf',
+        sky: '#00c2e0',
+        lime: '#51e898',
+        pink: '#ff78cb',
+        black: '#344563',
```

Comment:
For maintainability, let's extract these definitions into a seperate class and import that were it's needed